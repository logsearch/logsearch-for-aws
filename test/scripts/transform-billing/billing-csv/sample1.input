"InvoiceID","PayerAccountId","LinkedAccountId","RecordType","RecordID","BillingPeriodStartDate","BillingPeriodEndDate","InvoiceDate","PayerAccountName","LinkedAccountName","TaxationAddress","PayerPONumber","ProductCode","ProductName","SellerOfRecord","UsageType","Operation","RateId","ItemDescription","UsageStartDate","UsageEndDate","UsageQuantity","BlendedRate","CurrencyCode","CostBeforeTax","Credits","TaxAmount","TaxType","TotalCost"
"12345678","123456789012","","PayerLineItem","3000000000000000001","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AmazonCloudFront","Amazon CloudFront","Amazon Web Services, Inc.","JP-Requests-Tier1","GET","3506658","$0.0090 per 10,000 HTTP Requests","2015/07/01 00:00:00","2015/07/31 23:59:59","1.0",,"USD","0.01","0.0","0.000000","None","0.010000"
"12345678","123456789012","","PayerLineItem","3000000000000000002","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AWSDataTransfer","AWS Data Transfer","Amazon Web Services, Inc.","USW2-SAE1-AWS-In-Bytes",,"3510921","$0.00 per GB - US West (Oregon) data transfer from South America (Sao Paulo)","2015/07/01 00:00:00","2015/07/31 23:59:59","0.15887837",,"USD","0.00","0.0","0.000000","None","0.000000"
"12345678","123456789012","","PayerLineItem","3000000000000000003","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AmazonCloudFront","Amazon CloudFront","Amazon Web Services, Inc.","JP-Requests-Tier2-HTTPS",,"3506659","$0.012 per 10,000 HTTPS Requests","2015/07/01 00:00:00","2015/07/31 23:59:59","12585.0",,"USD","0.02","0.0","0.000000","None","0.020000"
"12345678","123456789012","","PayerLineItem","3000000000000000004","2015/07/01 00:00:00","2015/07/31 23:59:59","2015/08/03 11:51:47","John Doe","","123 Main St, San Jose, California, 95131, US",,"AWSDataTransfer","AWS Data Transfer","Amazon Web Services, Inc.","DataTransfer-Out-Bytes",,"3510871","$0.000 per GB - first 1 GB of data transferred out per month","2015/07/01 00:00:00","2015/07/31 23:59:59","0.02131482",,"USD","0.00","0.0","0.000000","None","0.000000"
