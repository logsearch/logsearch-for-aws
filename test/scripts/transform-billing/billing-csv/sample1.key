123456789012-aws-billing-detailed-line-items-2015-07.csv.zip
