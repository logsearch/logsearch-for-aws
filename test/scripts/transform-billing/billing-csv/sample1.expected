{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":0,"RecordType":"PayerLineItem","RecordID":"3000000000000000001","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AmazonCloudFront","ProductName":"Amazon CloudFront","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"JP-Requests-Tier1","Operation":"GET","RateId":3506658,"ItemDescription":"$0.0090 per 10,000 HTTP Requests","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":1.0,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.01,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.01}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":0,"RecordType":"PayerLineItem","RecordID":"3000000000000000002","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AWSDataTransfer","ProductName":"AWS Data Transfer","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"USW2-SAE1-AWS-In-Bytes","Operation":null,"RateId":3510921,"ItemDescription":"$0.00 per GB - US West (Oregon) data transfer from South America (Sao Paulo)","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":0.15887837,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.0,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.0}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":0,"RecordType":"PayerLineItem","RecordID":"3000000000000000003","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AmazonCloudFront","ProductName":"Amazon CloudFront","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"JP-Requests-Tier2-HTTPS","Operation":null,"RateId":3506659,"ItemDescription":"$0.012 per 10,000 HTTPS Requests","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":12585.0,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.02,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.02}
{"InvoiceID":12345678,"PayerAccountId":123456789012,"LinkedAccountId":0,"RecordType":"PayerLineItem","RecordID":"3000000000000000004","BillingPeriodStartDate":"2015/07/01 00:00:00","BillingPeriodEndDate":"2015/07/31 23:59:59","InvoiceDate":"2015/08/03 11:51:47","PayerAccountName":"John Doe","LinkedAccountName":null,"TaxationAddress":"123 Main St, San Jose, California, 95131, US","PayerPONumber":null,"ProductCode":"AWSDataTransfer","ProductName":"AWS Data Transfer","SellerOfRecord":"Amazon Web Services, Inc.","UsageType":"DataTransfer-Out-Bytes","Operation":null,"RateId":3510871,"ItemDescription":"$0.000 per GB - first 1 GB of data transferred out per month","UsageStartDate":"2015/07/01 00:00:00","UsageEndDate":"2015/07/31 23:59:59","UsageQuantity":0.02131482,"BlendedRate":null,"CurrencyCode":"USD","CostBeforeTax":0.0,"Credits":0.0,"TaxAmount":0.0,"TaxType":"None","TotalCost":0.0}
